go/ai/generator.go

```diff
@@ -76,6 +76,16 @@ func DefineGenerator(provider, name string, metadata *GeneratorMetadata, generat
 	return generator{a}
 }
 
+// LookupGenerator looks up a [Generator] registered by [DefineGenerator].
+// It returns nil if the Generator was not defined.
+func LookupGenerator(provider, name string) Generator {
+	action := core.LookupActionFor[*GenerateRequest, *GenerateResponse, *Candidate](core.ActionTypeModel, provider, name)
+	if action == nil {
+		return nil
+	}
+	return generator{action}
```

Comment:
For whatever reason I'm trying to avoid doing this additional wrapping... maybe I'm overreacting? Just tell me it's OK and I'll let it go. :)

Reply:
resolved on followup